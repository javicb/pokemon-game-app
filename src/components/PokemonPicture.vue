<template>
    <div class="pokemon-container">
        <img v-if="!showPokemon" id="hidden-pokemon" class="hidden-pokemon" alt="img-pokemon" :src="imgSrc">
        <img v-else id="pokemon" class="fade-in" alt="img-pokemon" :src="imgSrc">
    </div>
</template>

<script>
export default {
    name: 'PokemonPicture',
    props: {
        pokemonId: {
            type: Number,
            required: true,
            default: 0
        },
        showPokemon: {
            type: Boolean,
            requiered: true,
            default: false
        }
    },
    computed: {
        imgSrc() {
            return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/${this.pokemonId}.png`
        }
    },
}
</script>

<style scoped>
.pokemon-container {
    display: flex;
    height: 200px;
    justify-content: center;
}

img {
    -moz-user-select: none;
    -ms-user-select: none;
    -webkit-user-drag: none;
    -webkit-user-select: none;
    display: block;
    height: 21%;
    margin-left: auto;
    margin-right: auto;
    position: absolute;
    user-select: none;
}

.hidden-pokemon {
    filter: brightness(0);
}
</style>