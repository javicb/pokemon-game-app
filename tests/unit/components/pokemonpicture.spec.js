import { shallowMount } from '@vue/test-utils';
import PokemonPicture from '@/components/PokemonPicture.vue';

describe('PokemonPicture component', () => {

  test('should be match with snapshot', () => {

    // Arrange
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 1,
        showPokemon: false,
      },
    });

    // Assert
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('it should show the hidden image and the pokemonId 100', () => {

    // Arrange
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 100,
        showPokemon: false,
      },
    });

    // Act
    const [img1, img2] = wrapper.findAll('img');

    // Assert
    expect(img1.exists()).toBe(true);
    expect(img2).toBe(undefined);
    expect(img1.classes('hidden-pokemon')).toBe(true);
    expect(img1.attributes('src')).toBe(`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/100.png`);
  });

  test('should show the pokemon if showpokemon=true', () => {

    // Arrange
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 100,
        showPokemon: true,
      },
    });

    // Act
    const img = wrapper.find('img');

    // Assert
    expect(img.exists()).toBe(true);
    expect(img.classes('hidden-pokemon')).toBe(false);
    expect(img.classes('fade-in')).toBe(true);
  });
})