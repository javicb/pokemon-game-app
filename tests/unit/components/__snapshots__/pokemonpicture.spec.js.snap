// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPicture component should be match with snapshot 1`] = `<div class="pokemon-container"><img id="hidden-pokemon" class="hidden-pokemon" alt="img-pokemon" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png"></div>`;
